src/lib/storage/BaseRecord.ts

```diff
@@ -0,0 +1,35 @@
+export enum RecordType {
+  BaseRecord = 'BaseRecord',
+  ConnectionRecord = 'ConnectionRecord',
+  BasicMessageRecord = 'BasicMessageRecord',
+}
+
+export abstract class BaseRecord {
+  createdAt: number;
+  updatedAt?: number;
+  id: string;
+  tags: { [keys: string]: string };
+
+  // Required because Javascript doesn't allow accessing static types
+  // like instance.static_memeber
+  type: RecordType = RecordType.BaseRecord;
+  static type: RecordType = RecordType.BaseRecord;
```

Comment:
I'm sorry, but I still don't understand why we have both variants here? 🤔 😄

Reply:
That's because typescript converts a static member inside a class to be a property of the class.

```typescript
class Foo {
  static bar: string = "FOOBAR";
}
```

ends up being transpiled to:

```javascript
"use strict";
class Foo {
}
Foo.bar = "FOOBAR";
```

Unfortunately, we cannot access `bar` from an instance of `Foo` but only Foo itself.

I used the static member to allow passing a `typeof Class` to the constructor of Repository and allow initializations like `const repo = new Repository<ConnectionRepository>(ConnectionRecord, indyStorageInstance);`

At the same time, I need it to be accessible from an instance since I destructure it in `IndyStorageService::{save,update,delete}`

Reply:
Ok, I feel like it says us something, but we don't have to address this in this PR ;)